wiki/Bazel-Setup-Instructions-for-Mac.md

```diff
@@ -0,0 +1,117 @@
+
+**Note: Always use the Rosetta terminal for Bazel setup and running the `bash setup.sh` or any Bazel build-related commands.**
+
+### 2. Installing Bazel
+
+1. Install Bazel from [here](https://docs.bazel.build/versions/4.0.0/install-os-x.html#install-with-installer-mac-os-x). Make sure that you follow the instructions for installing a specific version (Oppia Android requires 4.0.0 and won't build on other versions).
+2. After Install Bazel from the above instructions, set the Bazel path permanently in your terminal using these commands:
+    ```
+    sudo nano /etc/paths
+    ```
+   - Enter your password, when prompted.
+   - Go to the bottom of the file, and enter this path (`/Users/{YourMacUserName}/bin`)
+   - Hit control-x to quit.
+   - Enter “Y” to save the modified buffer.
+   - That’s it!  To test it, in a new terminal window, type: `echo $PATH`
+    
+
+3. That’s it, now Bazel is setup permanently in your terminal, you can again check it by using command:
+   ```
+   bazel –version
+   ```
+
+**Note: The path for Bazel 4.0.0 is set before running Bazel build otherwise you get an error.**
+
+### 3. Installing the OpenJDK 8
+
+Oppia Android also requires OpenJDK 8.
+Follow the instructions [here](https://installvirtual.com/install-openjdk-8-on-mac-using-brew-adoptopenjdk/) to install OpenJDK 8. 
+Note that this requires the installation of brew as a pre-requisite, which can be done by following the instructions [here](https://mac.install.guide/homebrew/index.html). You can then set up your `$JAVA_HOME` environment variable using these [instructions](https://stackoverflow.com/a/75167958/11396524).
+
+
+### 4. Installing Python 2
+
+```
+brew install pyenv
+pyenv install 2.7.18
+pyenv global 2.7.18
+```
+
+- To make sure Python 2 is active in your environment, follow these commands in **oppia-android** directory before running the Bazel build.
+
+```
+export PATH="$(pyenv root)/shims:${PATH}"
+python --version
+```
+
+- To set the python path (`$(pyenv root)/shims:${PATH}`) permanently use the same commands as mentioned in **Step 2 Installing Bazel**.
+
+
+**Note: The path for Python 2  is set before running Bazel build otherwise you get an error.**
+
+
+### 5. Set the Android SDK
+
+- Ensure that your `ANDROID_HOME` environment variable is set to the location of your Android SDK before running Bazel build.
+
+- To do this, find the path to the installed SDK using Android Studio’s SDK Manager (install SDK 28). Assuming the SDK is installed to default locations, you can use the following command to set the `ANDROID_HOME` variable:
+```
+export ANDROID_HOME=$HOME/Library/Android/sdk
+```
+- To confirm that it is set, run the following command:
+```
+echo $ANDROID_HOME
+```
+
+To set the Android SDK path (`$HOME/Library/Android/sdk`) permanently use the same commands as mentioned in **Step 2 Installing Bazel**.
```

Comment:
I'm starting to wonder whether you should just have separate installation steps for each thing, and then have a single step that is "set the paths/variables permanently" and give a clear command that lets them do it all at once for all the dependencies above. Ideally all your steps are self-contained and don't involve referencing other steps.

Reply:
Thanks Sean for this suggestions, I added one separate and common step for all of these.

Reply:
Done